### ISSUE TITLE
Permitir retornar CEP's angolanos

### ISSUE THREAD

**Issue by acidiney** [acidiney/buscador-ao#2]
## Motivação
A ideia inicial no buscador-ao era retona nif's existentes em Angola. Porém com o andamento do projecto, decidi adicionar algumas funções que espero que sejam bem vistas por parte da comunidade angolana.

## Objetivo
Disponibilizar um endpoint para verificar um determinado CEP angolano e uma interface gráfica que as pessoas que não conhecem podem verificar o seu CEP por intermédio do `País`, `Província`, `Cidade` e `Bairro`.

### Ações a se tomar
 
- Mandar um e-mail para os correio de Angola para saber se existe e se podem disponibilizar essas informações.

- No caso, da primeira ação não funcionar, gerar automaticamente através de um algoritmo esses CEP e guardar eles aqui, para o pessoal avaliar.

- Feita a avaliação e a publicação do mesmo disponibilizar a release para todos poderem usar, num endpoint semelhante a este:
```/cep/:xxxx-xxxx```

## Conclusão.

Acredito que esse recurso seja uma mais valia para todos, não só os developers mais também o publico em geral que compra pela internet, ter uma API que retorna essas informações pode ajudar a TUPUCA por exemplo a identificar um endereço pelo CEP ou a Kubinga a fazer um melhor trajeto para onde o cliente quer receber a encomenda. 

E sim, eu sei que o nosso problema vai mais além do que isso, a questão do Google também entra nesta issue. Mas vamos tentar resolver cada coisa de uma vez? ^^

**Comment by JoseCage** [comment]
> Disponibilizar um endpoint para verificar um determinado CEP angolano e uma interface gráfica que as pessoas que não conhecem podem verificar o seu CEP por intermédio do País, Província, Cidade e Bairro.

Achei interessante. Mas então. A ideia é permitir que a plataforma tenha os **CEP's** que possam ser usados para facilitar na localização/mapping? E considerando que os correios não vá disponibilizar tal informação (e se tiverem está uma bagunça) a ideia é gerar identificadores de cada Província//Município//Bairro?

**Comment by acidiney** [comment]
Sim, já restou a redigir um texto para os correiro para saber se eles têm e se podem disponiblizar no caso de terem.

Sim no caso de não disponiblizar criar um identidicador nosso que possa ser usado para substituir a ausência deles.

**Comment by JoseCage** [comment]
> Sim no caso de não disponiblizar criar um identidicador nosso que possa ser usado para substituir a ausência deles.

:rocket:  Seria uma boa. Só fica a questão de adaptação e reconhecimento "pelas pessoas, empresas que possam usar"

**Comment by acidiney** [comment]
![image](https://user-images.githubusercontent.com/24493328/82440431-31d37380-9a94-11ea-9cfa-d57ce70f9356.png)
Eis que vou para mandar o email, e vejo o site dos correios assim.

**Comment by acidiney** [comment]
-- Email enviado aos correios de angola --
Bom dia caríssimos,
 
Chamo-me Acidiney Dias, recentemente criei um projecto que visa dar acesso aos desenvolvedores de algumas informações úteis no país, de forma a centralizar a informação em um único sitio e os demais desenvolvedores/empreendedores podem ter acesso a essa informação... Seja para contar como fonte fidedigna ou para fechar as lacunas nos projectos.
 
O projecto chama-se buscador e está disponível como forma de API ( um mecanismo de comunicação entre sistemas ).
 
Dada este pequeno briefing. Gostaria de saber com os correios de Angola se existe alguma espécie de código postal no país que não seja restrita para vocês que possa ser compartilha entre os demais que precisem dessa informação.
 
O meu motivo é simples, Aplicações como Tupuca, Kubinga, Garçon ou até mesmo a DHL... Nesse momento usam alternativas não muito certeiras para interagir com os pontos de entrega que o cliente introduziu. Por exemplo: se eu faço um pedido ao Tupuca eu tenho que usar o mapa do google ( o que não é mal ) para me localizar, porém se eu tiver algum código postal que se refere a zona em que estou a procura pelo local fica mais curta, porque ele vai delimitar uma zona de procura.
 
Por outra, quando encomendamos alguma coisa ( e digo encomendar, porque faço bastante ) tenho que literalmente ligar para lá e dizer que vivo no Luanda Sul de Viana  porque existe um do Talatona também, coisa que poderia ser evitada se passa-se de antemão o meu código postal.
 
Entendo que pode ser complicado neste momento implementar isso, mas acredito que para tudo existe um inicio, e que é melhor iniciar que ficar para trás.
Por isso, na eventualidade de não terem um serviço como este, gostaria de me oferecer para ajudar na implementação do mesmo, desta forma conseguimos resolver problemas de muitas pessoas, e vejo isso como um grande passo para as novas tecnologias.
 
Aguardo ansiosamente a vossa resposta,
 
 
Obrigado,
 
Acidiney Dias, Buscador.ao

-- fim do email ---

**Comment by acidiney** [comment]
Acho que também vou mandar uma a DHL, acredito que eles devem ter alguma coisa também.

**Comment by JoseCage** [comment]
Well done. :heavy_check_mark: 

O email está fixe.

Quanto a mandar para DHL também, acho válido levando em conta que embora muitas vezes trabalharem integrados aos correios, elem devem ter uma forma própria de "mapear" os locais.

**Comment by acidiney** [comment]
>Quanto a mandar para DHL também, acho válido levando em conta que embora muitas vezes trabalharem integrados aos correios, elem devem ter uma forma própria de  "mapear" os locais.

Sim, estou considerando isso mesmo.

**Comment by JoseCage** [comment]
> Sim, estou considerando isso mesmo.

Top. :rocket:

**Comment by acidiney** [comment]
![image](https://user-images.githubusercontent.com/24493328/83422522-9215ce00-a421-11ea-9cf2-5bf2612b2365.png)
http://viacep.com.br/ws/01310100/json

Exemplo que como pode funcionar

**Comment by acidiney** [comment]
Até agora não obtive nenhum feedback por parte dos correios/dhl ou até mesmo a FedEx.

Então vou marcar essa issue para sugestão de implementação, fazendo com que ela dependa da issue #3

**Comment by JoseCage** [comment]
> Exemplo que como pode funcionar

Boa.

Bora lá.

**Comment by JoseCage** [comment]
> Até agora não obtive nenhum feedback por parte dos correios/dhl ou até mesmo a FedEx.

Aqui as empresas não gostam de estar abertos à evolução. Infelizmente. :cry:
